server/src/distributed_engine/mod.rs

```diff
@@ -132,19 +132,18 @@ where
                 }
             };
             if status != 0 {
-                self.local_storage.create_directory(path)?;
                 return Err(EngineError::from(status));
             }
         }
-        Ok(())
+        self.local_storage.delete_directory(path.clone())
     }
 
     pub async fn read_dir(&self, path: String) -> Result<Vec<String>, EngineError> {
```

Comment:
So, you decide not to add size and offset to read_dir now?

Reply:
yes